### DISCUSSION TITLE
Errors in repeatable fields that supply arrays instead of multiple indexed fields

### DISCUSSION THREAD

**Discussion by zayenz** [airjp73/rvf#82] [Q&A] [answered]
Using a multi select input group for repeatable fields (as discussed in #80 and #81 and example in [this gist](https://gist.github.com/zayenz/b9e6ca74773481f8eeb827543850d27a)) named `fruits`, the errors supplied are for indexed fields such as `fruits[0]`. This makes it harder to write components, since the full form context needs to be parsed for errors instead of just using the error from `useField`. 

Is there a way to get errors to the field supplying the array?

**Comment by airjp73** [comment, marked as answer]
For now, I recommend using the [array syntax](https://www.remix-validated-form.io/arrays-and-nested) instead of repeated field names if you have per-input validation errors.

That said, returning an array of errors for `fruits` is something I would like to support at some point.

**Reply by zayenz** [reply]
In this case I have only a single field (react select multi input) that supplies all the values, so that is why it is without the array syntax. 

It is easy enough to parse it out locally and collate the errors, so not a higher issue. I was mostly wondering if there was any more idiomatic solution. But might be worth mentioning cases where a single input field supplies all the values in the documentation.

**Comment by sdriffill-tm** [comment]
Our team solution to this was the following:

```ts
import type { FieldErrors } from 'remix-validated-form';
import { useFormContext } from 'remix-validated-form';

function getFieldErrors(allFieldErrors: FieldErrors, name: string) {
  return Object.entries(allFieldErrors).reduce((errors, error) => {
    const [fieldName, errorMessage] = error;
    if (fieldName === name || fieldName.startsWith(`${name}[`)) {
      return [...errors, errorMessage];
    }
    return errors;
  }, [] as string[]);
}

const { fieldErrors: allFieldErrors } = useFormContext();
const fieldErrors = getFieldErrors(allFieldErrors, name);
```

Then you can choose what to do with the array of errors - in our case we display each on a new line under the input.
